frontend/src/lib/lemon-ui/LemonDropdown/LemonDropdown.tsx

```diff
@@ -39,16 +39,16 @@ export const LemonDropdown: React.FunctionComponent<LemonDropdownProps & React.R
             ref
         ) => {
             const [, parentPopoverLevel] = useContext(PopoverOverlayContext)
-            const [localVisible, setLocalVisible] = useState(false)
+            const [localVisible, setLocalVisible] = useState(visible ?? false)
 
             const floatingRef = useRef<HTMLDivElement>(null)
             const referenceRef = useRef<HTMLSpanElement>(null)
 
             const effectiveVisible = visible ?? localVisible
 
             useEffect(() => {
-                onVisibilityChange?.(effectiveVisible)
-            }, [effectiveVisible, onVisibilityChange])
+                onVisibilityChange?.(localVisible)
+            }, [localVisible, onVisibilityChange])
```

Comment:
This allows the component to be controlled whilst also indicating to the parent that the state has changed